### DISCUSSION TITLE
Is it possible to parse texts directly instead of "send_commands" to get output ?

### DISCUSSION THREAD

**Discussion by zhangineer2** [carlmontanari/scrapli#340] [General]
Hello

I'd like to just parse commands output, instead of SSH to a device, send a command and parse the output. Reason being that I don't always have direct access to these devices.

So maybe a way to "mock" the command outputs ? I would still provide the device/Driver information.

Thanks
Peter

**Comment by carlmontanari** [comment]
if you just mean parse via textfsm/genie/ttp, there are helper funcs [here](https://github.com/carlmontanari/scrapli/blob/main/scrapli/helper.py) but this isn't really a scrapli question/problem since we are just using those other parsers/libraries anyway

**Comment by zhangineer2** [comment]
@carlmontanari  thank you for the late reply and sorry that I opened this as "issue". 

so I looked into the helper and tried the following, but the template was not found.  Thoughts ?

```
from scrapli.helper import _textfsm_get_template, genie_parse, textfsm_parse, ttp_parse
from typing import TYPE_CHECKING, Any, Dict, Iterable, List, Optional, TextIO, Union, cast
from io import TextIOWrapper

template = _textfsm_get_template(
    platform='cisco_nxos', command='show run'
)
```

**Reply by carlmontanari** [reply]
No problem. There’s definitely no template for parsing show run. Go check out ntc templates and/or write your own (or check out ttp)

**Reply by zhangineer2** [reply]
duh..my bad. you are right, there wouldn't be a template for a `show run` command, it was a typo on my part. with `show version` it  worked just fine.

- if I were to build my own template, I am assuming that I would create a new local directory, store my own template there, and pass in the `template` parameter specifically when calling the private function `_textfsm_get_template` , correct ? 
- the alternative would be to contribute to the ntc-template repo (which may take longer to get approved)
